osgar/drivers/rosmsg.py

```diff
@@ -294,9 +297,13 @@ def slot_raw(self, timestamp, data):
             if prev != self.timestamp_sec:
                 self.bus.publish('sim_time_sec', self.timestamp_sec)
 
-            ms = self.timestamp_nsec//1000000
-            if self.timestamp_sec > 0 and ms % 50 == 0:  # 20Hz
+            now = timedelta(seconds=self.timestamp_sec,
+                            milliseconds=self.timestamp_nsec//1000000)
+            if self.timestamp_sec > 0 and (
+                    self.last_cmd_timestamp is None or
+                    now - self.last_cmd_timestamp >= timedelta(milliseconds=48)):  # ~20 Hz
```

Comment:
this is wrong place to be ... I had some feeling that "odometry" was important in rosproxy and that was the reason. This is some post-mix-goulash cause by integration of MOBoS(?) where odometry was not available or was not stamped so time was used. In Virtual SubT the clock is very rare, down to 1second.

Reply:
```
0:00:07.738796 45 [48, 0]
0:00:07.739480 46 b'cmd_vel 0.300000 -1.193805'
0:00:07.741236 46 b'broadcast \x93\x05\xacsim_time_sec0'
0:00:07.742133 46 b'broadcast \x93\x06\xa3xyz\x920\x93\xcb\xc0\x0b\xff\xe6O\x18\x9c\x11\xcb@\x10\x00\rK\x90g\x87\xcb?\xbe\x14y\xbbne\x08'
0:00:08.098952 46 b'broadcast \x93\x07\xaatrace_info\x82\xa5A900L\x91\x92/0\xa5B900R\x91\x92//'
0:00:08.118294 46 b'broadcast \x93\x08\xaatrace_info\x82\xa5A900L\x91\x92/0\xa5B900R\x91\x92/0'
0:00:08.119392 46 b'broadcast \x93\t\xabrobot_trace\x81\xa5B900R\x91\x920\x93\xcb\xc0\x0b\xff\xfe\x85D\xf4\xde\xcb?\xff\xfe\x00\xd8(\\\xb3\xcb?\xbe\x17 \xae\x9fm\x91'
0:00:17.147757 45 [49, 0]
0:00:17.148577 46 b'cmd_vel 1.428000 0.062832'
0:00:17.149439 46 b'broadcast \x93\n\xacsim_time_sec1'
0:00:17.151515 46 b'broadcast \x93\x0b\xa3xyz\x921\x93\xcb\xc0\t\x97\xe7\x94\xa8\xae\xeb\xcb@\x0f\xd6\x10\xa2L,\x8e\xcb?\xbe\x1a\xcb~\xf1\xe7\xe2'
```

Reply:
I would vote for extra fix independent of this PR, which would publish `/<ROBOT_NAME>/cmd_vel` via `cloudsim2osgar.py`

Reply:
Experiment in #800